assets/js/blocks/attribute-filter/block.js

```diff
@@ -168,16 +169,11 @@ const AttributeFilterBlock = ( {
 	] );
 
 	// Track checked STATE changes - if state changes, update the query.
-	useEffect(
-		() => {
-			if ( ! blockAttributes.showFilterButton ) {
-				onSubmit();
-			}
-		},
-		// There is no need to add blockAttributes.showFilterButton as a dependency.
-		// It will only change in the editor and there we don't need to call onSubmit in any case.
-		[ checked, onSubmit ]
-	);
+	useEffect( () => {
+		if ( ! blockAttributes.showFilterButton ) {
+			onSubmit( checked );
+		}
+	}, [ blockAttributes.showFilterButton, checked, onSubmit ] );
```

Comment:
Missing dep in effect. `blockAttributes.showFilterButton` is not expected to change during the lifecycle of the component, so it should be safe.